server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesAggregator.java

```diff
@@ -42,7 +42,7 @@ public TimeSeriesAggregator(
         CardinalityUpperBound bucketCardinality,
         Map<String, Object> metadata
     ) throws IOException {
-        super(name, factories, context, parent, bucketCardinality, metadata);
+        super(name, factories, context, parent, CardinalityUpperBound.MANY, metadata);
```

Comment:
why is this hardcoded?

Reply:
I think it should be `MANY` actually. Though this isn't related and I should have caught it earlier. `MANY` here means "my children will an unbounded number of buckets". It's normal to do stuff like `bucketCardinality.multiply(filters.size())` if, say, you were filters and knew precisely how many buckets you'd make. But we never do.

Reply:
This is actually an unrelated change - I'll back it out.  It's necessary for the rate agg to work but I'm not sure that it's the correct way to fix things.

Reply:
FWIW I believe it's correct. I don't care if you keep it or take it out and do it in the rate agg.